Fix major sonar issues

rozdy opened the pull request.

niksv commented on services/src/northbound-service/northbound/src/main/java/org/openkilda/northbound/service/impl/FlowServiceImpl.java:
```diff
@@ -464,7 +464,8 @@ private CommandMessage buildDeleteFlowCommand(final String id, final String corr
     private FlowPathPayload buildFlowPathPayload(List<FlowPathDto> paths, String flowId) {
-        FlowPathDto askedPathDto = paths.stream().filter(e -> e.getId().equals(flowId)).findAny().get();
+        FlowPathDto askedPathDto = paths.stream().filter(e -> e.getId().equals(flowId)).findAny()
+                .orElseThrow(() -> new IllegalStateException(format("Flow %s not found.", flowId)));
```
incorrect message. Here we didn't found **path** for flow

rozdy replied:
Fixed.

niksv commented on services/wfm/src/main/java/org/openkilda/wfm/topology/flow/service/FlowService.java:
```diff
@@ -959,12 +956,6 @@ private FlowPath buildFlowPath(Flow flow, Path path, FlowPathStatus pathStatus,
         return flowPath;
     }
 
-    private void ensureEncapsulationType(Flow flow) {
```
why it was removed?

rozdy commented on services/src/atdd-staging/src/main/java/org/openkilda/atdd/staging/steps/FlowSteps.java:
```diff
@@ -351,12 +352,13 @@ private int getAllowedVlan(Set<FlowPayload> flows, SwitchId switchDpId) {
         return availableVlansRange.asRanges().stream()
                 .flatMap(range -> ContiguousSet.create(range, DiscreteDomain.integers()).stream())
-                .findFirst().get();
+                .findFirst().orElseThrow(() -> new IllegalStateException("No ranges found."));
```
Fixed

rozdy commented:
@surabujin 

> > Reviewed 17 of 17 files at r1.
> > _[Reviewable](https://reviewable.io/reviews/telstra/open-kilda/2862)_ status: all files reviewed, 5 unresolved discussions (waiting on @dpoltavets, @nikitamarchenko, @rozdy, and @sergii-iakovenko)
> 
> _[services/src/floodlight-service/floodlight-modules/src/main/java/org/openkilda/floodlight/switchmanager/SwitchManager.java, line 706 at r1](https://reviewable.io/reviews/telstra/open-kilda/2862#-LrNirj01xaGgIz6XMNf:-LrNirj1CL_s_wmqb3Ip:budj8v7) ([raw file](https://github.com/telstra/open-kilda/blob/9a96a2e4a87c0de2e42a506c5d7e25eaa6c893be/services/src/floodlight-service/floodlight-modules/src/main/java/org/openkilda/floodlight/switchmanager/SwitchManager.java#L706)):_
> 
> > ```java
> >         } catch (ExecutionException | InterruptedException | TimeoutException e) {
> >             logger.error("Could not get flow stats for {}.", dpid, e);
> >             if (e instanceof InterruptedException) {
> > ```
> 
> em... why? Make one more `catch` block, dedicated to `InterruptException`.

Fixed.

> 
> _[services/src/kilda-core/kilda-model/src/main/java/org/openkilda/model/Flow.java, line 141 at r1](https://reviewable.io/reviews/telstra/open-kilda/2862#-LrNjGLC-tE7UC2ThTCG:-LrNjGLC-tE7UC2ThTCH:bb7r2f0) ([raw file](https://github.com/telstra/open-kilda/blob/5192df95f7388b9e003f14338fdeadfb20d1600f/services/src/kilda-core/kilda-model/src/main/java/org/openkilda/model/Flow.java#L141)):_
> 
> > ```java
> >     private boolean periodicPings;
> > 
> >     @NonNull
> > ```
> 
> why?

Because it can be null after mapping from FlowCreate/FlowUpdate requests. We need either add NonNull to builder and move this logic inside corresponding mapper

`if (flow.getEncapsulationType() == null) {
            flow.setEncapsulationType(kildaConfigurationRepository.get().getFlowEncapsulationType());
        }`

or remove NonNull here.

> 
> _[services/src/kilda-pce/src/main/java/org/openkilda/pce/finder/BestCostAndShortestPathFinder.java, line 347 at r1](https://reviewable.io/reviews/telstra/open-kilda/2862#-LrNj_mmBenZMiX0UYfP:-LrNj_mmBenZMiX0UYfQ:bb76mu4) ([raw file](https://github.com/telstra/open-kilda/blob/5192df95f7388b9e003f14338fdeadfb20d1600f/services/src/kilda-pce/src/main/java/org/openkilda/pce/finder/BestCostAndShortestPathFinder.java#L347)):_
> 
> > ```java
> >         for (Edge i : path) {
> >             Node srcSwitch = i.getSrcSwitch();
> >             if (srcSwitch == null) {
> > ```
> 
> Why?

Switch src can't be null here.

> 
> _[services/src/kilda-pce/src/main/java/org/openkilda/pce/model/Node.java, line 135 at r1](https://reviewable.io/reviews/telstra/open-kilda/2862#-LrNk4s81WIZpE73JPsb:-LrNk4s81WIZpE73JPsc:b-sg6bu1) ([raw file](https://github.com/telstra/open-kilda/blob/9a96a2e4a87c0de2e42a506c5d7e25eaa6c893be/services/src/kilda-pce/src/main/java/org/openkilda/pce/model/Node.java#L135)):_
> 
> > ```java
> >      * Restore node.
> >      */
> >     public void restore() {
> > ```
> 
> Is it used anywhere? Why we do not catch NPE during it's usage now?

Yes, it's used in InMemoryPathComputer. NPE can't be thrown here due to shortcut evaluation of a complex conditions.

> 
> _[services/wfm/src/main/java/org/openkilda/wfm/topology/ping/bolt/ResultDispatcher.java, line 47 at r1](https://reviewable.io/reviews/telstra/open-kilda/2862#-LrNlByc3vhIupsoKwjG:-LrNlByc3vhIupsoKwjH:bokmou) ([raw file](https://github.com/telstra/open-kilda/blob/9a96a2e4a87c0de2e42a506c5d7e25eaa6c893be/services/wfm/src/main/java/org/openkilda/wfm/topology/ping/bolt/ResultDispatcher.java#L47)):_
> 
> > ```java
> >         switch (kind) {
> >             case PERIODIC:
> >                 return STREAM_PERIODIC_ID;
> > ```
> 
> As for me... it is ugly. One obvious return statement at the method end is better than several in the middle.

Fixed back.

rozdy commented on services/src/floodlight-service/floodlight-modules/src/main/java/org/openkilda/floodlight/switchmanager/SwitchManager.java:
```diff
@@ -701,9 +701,13 @@ private OFInstructionMeter buildMeterInstructionForUnicastVxlanRule(IOFSwitch sw
             logger.error("Could not get flow stats for {}.", dpid, e);
             throw new SwitchNotFoundException(dpid);
+        } catch (InterruptedException e) {
+            logger.error("Could not get a barrier reply for {}.", sw.getId(), e);
```
Copy-paste error, fixed.

rozdy commented on services/src/kilda-core/kilda-model/src/main/java/org/openkilda/model/Flow.java:
```diff
@@ -138,7 +138,6 @@
     @Property(name = "periodic_pings")
     private boolean periodicPings;
 
-    @NonNull
```
Fixed.

rozdy commented on services/src/messaging/src/test/java/org/openkilda/messaging/info/flow/FlowStatusResponseTest.java:
```diff
@@ -29,8 +28,8 @@
-    private final String json = "{" +
-            "\"clazz\":\"org.openkilda.messaging.info.InfoMessage\"," +
+    private final String json = "{"
+            + "\"clazz\":\"org.openkilda.messaging.info.InfoMessage\"," +
```
Fixed.
